packages/runtime/container-runtime/src/gcSweepReadyUsageDetection.ts

```diff
@@ -0,0 +1,130 @@
+        localStorageImpl?: Pick<Storage, "getItem" | "setItem">,
+    ) {
+        const noopStorage = { getItem: () => null, setItem: () => {} };
+        if (localStorageImpl !== undefined) {
+            this.localStorage = localStorageImpl;
+        } else {
+            try {
+                // localStorage is not defined in Node environment so this throws
+                this.localStorage = localStorage ?? noopStorage;
+            } catch (error) {
+                this.localStorage = noopStorage;
+            }
+        }
+
+        if (this.localStorage === noopStorage) {
+            // This means the Blackout Period logic will not work.
+            this.mc.logger.sendTelemetryEvent({ eventName: "SweepReadyUsageDetectionHandlerNoOpStorage" });
+        }
+    }
+
+    /**
+      * If SweepReady Usage Detection is enabled, close the main container.
+      * If the "Blackout Period" is set, don't close the container more than once in that period.
+      *
+      * Once Sweep is fully implemented, this will be removed since the objects will be gone
+      * and errors will arise elsewhere in the runtime
+     */
+    public usageDetectedInMainContainer(errorProps: ITelemetryProperties) {
+        if (!sweepReadyUsageDetectionSetting.read(this.mc.config).mainContainer) {
+            return;
+        }
+
+        const pastClosuresMap: Record<string, { lastCloseTime: number; } | undefined> = (() => {
+            try {
+                const rawValue = this.localStorage.getItem(closuresStorageKey);
+                return rawValue === null
+                    ? {}
+                    : JSON.parse(rawValue) as Record<string, { lastCloseTime: number; } | undefined>;
+            } catch (e) {
+                return {};
+            }
+        })();
+
+        const lastCloseTime = pastClosuresMap[this.uniqueContainerKey]?.lastCloseTime;
+        const blackoutPeriodDays = this.mc.config.getNumber(blackoutPeriodDaysKey);
+
+        const error = new SweepReadyUsageError(
+            "SweepReady object used in Non-Summarizer Client",
+            { errorDetails: JSON.stringify({ ...errorProps, lastCloseTime, blackoutPeriodDays }) },
+        );
+
+        // Should close if...
+        const shouldClose = lastCloseTime === undefined // ...We've not closed before, or...
+            || blackoutPeriodDays === undefined // ...there's no blackout duration set, or...
+            || Date.now() > lastCloseTime + blackoutPeriodDays * oneDayMs; // ...we've passed the blackout period
+
+        if (shouldClose) {
+            // Update closures map in localStorage before closing
+            pastClosuresMap[this.uniqueContainerKey] = { lastCloseTime: Date.now() };
+            this.localStorage.setItem(closuresStorageKey, JSON.stringify(pastClosuresMap));
+
+            this.closeFn(error);
+        } else {
+            this.mc.logger.sendErrorEvent({ eventName: "IgnoringSweepReadyObjectUsage" }, error);
```

Comment:
I would prefer SweepReadyObjectUsage

Reply:
Well we already have the other logs.  Although those only go on Loaded - maybe we should update that to always log even for main container, and it's up to us to filter out to get better results.

@agarwal-navin thoughts?

Reply:
I think this is fine for now. As a follow up, we should probably move all the inactive / sweep ready logging logic here and then we can remove duplicates.
Maybe `SweepReadyObject_Used` to make the part after _ unique?

Reply:
Ok sure I can match the format of the other logs, good idea.